### PR TITLE
Add support for hexadecimal literals.

### PR DESCRIPTION
**Pull request by petermattis**
Changed the type of sqlSymType.ival from int to int64. We require
integers to be 64-bit, so be explicit about it.

### CODE REVIEW

File: sql/parser/scan_test.go [hunk 1]
```diff
@@ -267,9 +274,10 @@ func TestScanError(t *testing.T) {
 		sql string
 		err string
 	}{
-		{`1e`, "invalid floating point constant"},
-		{`1e-`, "invalid floating point constant"},
-		{`1e+`, "invalid floating point constant"},
+		{`1e`, "invalid floating point literal"},
+		{`1e-`, "invalid floating point literal"},
+		{`1e+`, "invalid floating point literal"},
+		{`0x`, "invalid hexadecimal literal"},
```
Comment (MBerhault) [inline sql/parser/scan_test.go]: Looking at the code, it looks like they will fail just fine, but can you add things like `0x0x...` and `00x...` just to make sure?

**Comment by MBerhault** [comment]
LGTM

File: sql/parser/scan_test.go (hunk 1 above)
Comment (petermattis) [inline sql/parser/scan_test.go]: Sure. Done. Did you take a look at both commits? I added a second commit which adds support for hex/octal/unicode string escapes.

Comment (petermattis) [inline sql/parser/scan_test.go]: Oh, never mind. I see I did that in a separate branch.

**Pull request merged**